staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_discovery_test.go

```diff
@@ -179,26 +183,38 @@ func TestRemoveAPIService(t *testing.T) {
 		}
 	}
 
-	aggregatedManager := apiserver.NewDiscoveryManager(aggyService)
-
-	for _, s := range apiServices {
-		aggregatedManager.AddAPIService(s, service)
-	}
+	testCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	testCtx, _ := context.WithCancel(context.Background())
+	aggregatedManager := apiserver.NewDiscoveryManager(aggyService)
 	go aggregatedManager.Run(testCtx.Done())
 
 	for _, s := range apiServices {
-		aggregatedManager.RemoveAPIService(s.Name)
+		aggregatedManager.AddAPIService(s, service)
 	}
 
-	cache.WaitForCacheSync(testCtx.Done(), aggregatedManager.ExternalServicesSynced)
+	require.True(t, cache.WaitForCacheSync(testCtx.Done(), aggregatedManager.ExternalServicesSynced))
```

Comment:
CI is currently failing because the test is stuck waiting at this line for 3 minutes (until testCtx.Done()).

Reply:
/cc @Jefftree

Reply:
Looks like there is a deadlock due to the APIService returning a 304 Not Modified, the aggregator takes the same lock twice. Opening up a follow-up PR to fix